src/main/java/com/iri/dao3/dao/jdbc/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+                car = constructCarFromResultSet(resultSet).get();
+            }
+            car.setDrivers(getAllDriversByCar(car));
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve car in method get", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectQuery = "SELECT c.id as id, c.model as model, "
+                + "m.id as manufacturer_id, m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturer m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(constructCarFromResultSet(resultSet).get());
+            }
+            for (Car car : cars) {
+                car.setDrivers(getAllDriversByCar(car));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't fetch cars in method getAll", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted != TRUE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            removeDriversFromCar(car);
+            for (Driver driver : car.getDrivers()) {
+                addDriverToCar(driver, car);
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't refresh car in method update", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't mark car in method delete", e);
+        }
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
```

Comment:
here we open many connections and statements because we use a loop
let's open them once and then inside the try block just execute the statement for each driver
```suggestion 
    private void addDriverToCar(Car car) {
```

Reply:
done